src/main/java/com/epam/aidial/core/storage/ResourceDescription.java

```diff
@@ -123,6 +123,15 @@ public static ResourceDescription fromDecoded(ResourceDescription description, S
         return fromDecoded(description.getType(), description.getBucketName(), description.getBucketLocation(), relativePath);
     }
 
+    public static ResourceDescription fromLink(String link) {
```

Comment:
Please check indexes there might be out of index exception.

Reply:
added